### PR TITLE
[Arm64] Support two GC attributes in pair forms

### LINKED ISSUE #11241: [Arm64 JIT] Support mismatched GC attr in ldp/stp
Currently when we emit ldp/stp we only get to provide one attr

This prevents usage of ldp/stp in a number of cases

We want to support mismatched GC attr for ldp/stp.

@dotnet/arm64-contrib @dotnet/jit-contrib

### PR DESCRIPTION
**Pull request by sdmaclea**
Also use in genCodeForCpObj

Fixes #11241

### CODE REVIEW

**Comment by sdmaclea** [comment]
@briansull PTAL

I have the patch for "[Arm64] Use ldp for mismatched GC types in genPutArgStk", but since the other patch has not yet merged I did not include it here.

**Comment by sdmaclea** [comment]
@briansull Since @BruceForstall merged the genPutArgStk PR.  I forced pushed this one to include that fix too.

**Comment by BruceForstall** [comment]
cc @dotnet/arm64-contrib @dotnet/jit-contrib

File: src/jit/emit.h [hunk 1]
```diff
@@ -655,6 +655,9 @@ class emitter
         // unnecessarily since the GC-ness of the second register is only needed for call instructions.
         // The instrDescCGCA struct's member keeping the GC-ness of the first return register is _idcSecondRetRegGCType.
         GCtype _idGCref : 2; // GCref operand? (value is a "GCtype")
+#ifdef _TARGET_ARM64_
+        GCtype _idGCref2 : 2; // GCref operand fir register 2? (value is a "GCtype")
```
Comment (briansull) [inline src/jit/emit.h]: spelling: fir => for (and remove ?)

File: src/jit/emit.h [hunk 2]
```diff
@@ -764,7 +767,7 @@ class emitter
         // x86:   24 bits
         // amd64: 16 bits
         // arm:   14 bits
-        // arm64: 14 bits
+        // arm64: 12 bits
 
         unsigned _idSmallCns : ID_BIT_SMALL_CNS;
```
Comment (briansull) [inline src/jit/emit.h]: We don't want the value of ID_BIT_SMALL_CNS to go below 12 bits.
It OK now, but we can't steal anymore bits from it on ARM64.

File: src/jit/codegenarm64.cpp [hunk 3]
```diff
@@ -3487,31 +3487,29 @@ void CodeGen::genCodeForCpObj(GenTreeObj* cpObjNode)
...
-                emit->emitIns_R_R_I(INS_str, attr, tmpReg, REG_WRITE_BARRIER_DST_BYREF, TARGET_POINTER_SIZE,
-                                    INS_OPTS_POST_INDEX);
-            }
+            emitAttr attr0 = emitTypeSize(compiler->getJitGCType(gcPtrs[i + 0]));
+            emitAttr attr1 = emitTypeSize(compiler->getJitGCType(gcPtrs[i + 1]));
+
+            emit->emitIns_R_R_R_I(INS_ldp, attr0, tmpReg, tmpReg2, REG_WRITE_BARRIER_SRC_BYREF, 2 * TARGET_POINTER_SIZE,
+                                  INS_OPTS_POST_INDEX, attr1);
+            emit->emitIns_R_R_R_I(INS_stp, attr0, tmpReg, tmpReg2, REG_WRITE_BARRIER_DST_BYREF, 2 * TARGET_POINTER_SIZE,
+                                  INS_OPTS_POST_INDEX, attr1);
```
Comment (briansull) [inline src/jit/codegenarm64.cpp]: From reading `emitInsMayWriteMultipleRegs` if appears that we only need/use the second attr1 value for ldp instructions

File: src/jit/emit.h (hunk 1 above)
Comment (sdmaclea) [inline src/jit/emit.h]: I'll remove the whole comment as it is redundant with the comment above.

File: src/jit/emit.h (hunk 2 above)
Comment (sdmaclea) [inline src/jit/emit.h]: It did seem unfortunate.

File: src/jit/codegenarm64.cpp (hunk 3 above)
Comment (sdmaclea) [inline src/jit/codegenarm64.cpp]: I noticed that as well.  It did seem disturbing.  I can remove the second attribute if you want (only the size of the first is actually used).  I only added it for symmetry

File: src/jit/emit.h (hunk 2 above)
Comment (sdmaclea) [inline src/jit/emit.h]: I was looking for places to free up bits.  The most obvious is the six bits used for instruction options.  I would guess these could be encoded in an enum to use less space.

File: src/jit/codegenarm64.cpp (hunk 3 above)
Comment (briansull) [inline src/jit/codegenarm64.cpp]: I think that it is fine how you have it.

File: src/jit/emitarm64.cpp [hunk 4]
```diff
@@ -5347,6 +5348,22 @@ void emitter::emitIns_R_R_R_I(instruction ins,
     id->idReg2(reg2);
     id->idReg3(reg3);
 
+    if (attrReg2 != EA_UNKNOWN)
+    {
+        if (EA_IS_GCREF(attrReg2))
```
Comment (briansull) [inline src/jit/emitarm64.cpp]: You could add this assert here:

assert((fmt == IF_LS_3B) || (fmt == IF_LS_3C));

**Pull request merged**